server/cluster/cluster_test.go

```diff
@@ -879,14 +881,20 @@ func (s *testClusterInfoSuite) TestOfflineAndMerge(c *C) {
 		regions = core.SplitRegions(regions)
 	}
 	heartbeatRegions(c, cluster, regions)
+	// wait for background job goroutine to handle the stats task.
+	time.Sleep(time.Second)
 	c.Assert(cluster.GetOfflineRegionStatsByType(statistics.OfflinePeer), HasLen, len(regions))
 
 	// Merge.
 	for i := 0; i < n; i++ {
 		regions = core.MergeRegions(regions)
 		heartbeatRegions(c, cluster, regions)
+		// wait for background job goroutine to handle the stats task.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I am concerned this kind of waiting is unstable, which will make the test fail in an unstable test environment. 🤔 Though if there is no other way to do this, I'm ok with it.

Reply:
Yes. But I have no idea about it.